### ISSUE TITLE
Destructuring array as function parameters results in a type error

### ISSUE THREAD

**Issue by niieani** [Microsoft/TypeScript#9394]
**TypeScript Version:**  2.0.0-dev.20160628 (the same error in all older versions)

**Code**

```ts
function test(a: string, b: {}, c: number) {}

test('one', {}, 3); // OK

let params = ['one', {}, 3];
test(...params); // ERROR
```

**Expected behavior:**
The destructured array should resolve to the same types for each of the parameters and should not give an error.

**Actual behavior:** 
Error: 

> Supplied parameters do not match any signature of call target.

**Comment by DanielRosenwasser** [comment]
Looks like a dupe of #5296.
